src/bun.js/child_process.exports.js

```diff
@@ -1,3 +1,995 @@
+  return Object.prototype.hasOwnProperty.call(obj, prop);
+}
+
+function ArrayPrototypePush(array, ...items) {
+  return array.push(...items);
+}
+
+function ArrayPrototypeReduce(array, callback, initialValue) {
+  return array.reduce(callback, initialValue);
+}
+
+function ArrayPrototypeJoin(array, separator) {
+  return array.join(separator);
+}
+
+function ArrayPrototypeMap(array, callback) {
+  return array.map(callback);
+}
+
+function ArrayPrototypeSlice(array, begin, end) {
+  return array.slice(begin, end);
+}
+
+function ArrayPrototypeFilter(array, callback) {
+  return array.filter(callback);
+}
+
+function ArrayPrototypeIncludes(array, searchElement, fromIndex) {
+  return array.includes(searchElement, fromIndex);
+}
+
+function ArrayIsArray(arg) {
+  return Array.isArray(arg);
+}
+
+function isArrayBufferView(value) {
+  return value instanceof ArrayBufferView;
+}
+
+//------------------------------------------------------------------------------
+// 6. Node error polyfills
+//------------------------------------------------------------------------------
+
+function ERR_INVALID_ARG_TYPE(name, type, value) {
+  return new Error(
+    `The argument '${name}' is invalid. Received '${value}' for type '${type}'`
+  );
+}
+
+function ERR_INVALID_ARG_VALUE(name, value, reason) {
+  return new Error(
+    `The value '${value}' is invalid for argument '${name}'. Reason: ${reason}`
+  );
+}
+
+// TODO: Add actual proper error implementation here
+function errnoException(err, name) {
+  return new Error(`Error: ${name}. Internal error: ${err.message}`);
+}
+
+//------------------------------------------------------------------------------
+// 7. Node stream polyfills
+//------------------------------------------------------------------------------
+Readable.prototype.on = function (event, listener) {
+  EventEmitter.prototype.on.call(this, event, listener);
+  if (event === "data") {
+    this._readableState.flowing = true;
+    this._read();
+  }
+};
+/**
+ * @param {ReadableStream} readableStream
+ * @param {{
+ *   highWaterMark? : number,
+ *   encoding? : string,
+ *   objectMode? : boolean,
+ *   signal? : AbortSignal,
+ * }} [options]
+ * @returns {Readable}
+ */
+export function newStreamReadableFromReadableStream(
+  readableStream,
+  options = {}
+) {
+  if (!isReadableStream(readableStream)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      "readableStream",
+      "ReadableStream",
+      readableStream
+    );
+  }
+
+  validateObject(options, "options");
+  const { highWaterMark, encoding, objectMode = false, signal } = options;
+
+  if (encoding !== undefined && !Buffer.isEncoding(encoding))
+    throw new ERR_INVALID_ARG_VALUE(encoding, "options.encoding");
+  validateBoolean(objectMode, "options.objectMode");
+
+  const reader = readableStream.getReader();
+
+  let closed = false;
+
+  const readable = new Readable({
+    objectMode,
+    highWaterMark,
+    encoding,
+    signal,
+
+    read() {
+      reader
```

Comment:
Instead of calling `reader.read()` here, let's use the undocumented faster  bun-specific API  `reader.readMany`. Note that `readMany` returns potentially a Promise or an object same-tick

I believe there is a way to push multiple values at once to a `Readable` instance

https://github.com/oven-sh/bun/blob/cf46ebdab37a126fefe291e14b5a2597cecbda06/src/bun.js/builtins/js/ReadableStreamInternals.js#L2122-L2156

Reply:
Actually, this will become irrelevant if we switch to an `onData` callback.

Reply:
Oh that's really nice. Yeah if we use the onData callback it may work. Although in some cases in userland users may opt to read via the `.read()` method of `Readable` , and technically should be able to pause the stream and resume at any point. See: `https://nodejs.org/api/stream.html#event-readable`